### PR TITLE
[design] pipe workflow error logs

### PR DESCRIPTION
**Pull request by Yancey0623**

### CODE REVIEW

File: doc/design/argo_workflow_on_sqlflow.md [hunk 1]
```diff
@@ -186,3 +186,27 @@ where
       }
       return responses, nil
     }
+
```
Comment (typhoonzero) [inline doc/design/argo_workflow_on_sqlflow.md:190]: No ending "```"

File: doc/design/argo_workflow_on_sqlflow.md [hunk 2]
```diff
@@ -186,3 +186,27 @@ where
...
+## Retrieve Error Logs From a failed Workflow
```
Comment (typhoonzero) [inline doc/design/argo_workflow_on_sqlflow.md]: failed -> Failed. And Success jobs also need to get the job training outputs?

File: doc/design/argo_workflow_on_sqlflow.md [hunk 3]
```diff
@@ -186,3 +186,27 @@ where
...
+
+A workflow maybe failed for many reasons, these error usually occur in the following two phase:
```
Comment (typhoonzero) [inline doc/design/argo_workflow_on_sqlflow.md]: maybe failed -> may fail

File: doc/design/argo_workflow_on_sqlflow.md [hunk 4]
```diff
@@ -186,3 +186,27 @@ where
...
+
+- The one is running a workflow, .e.g. the SQL syntax error.
```
Comment (typhoonzero) [inline doc/design/argo_workflow_on_sqlflow.md]: One phase is .... the other phase is ...

Comment (typhoonzero) [reply doc/design/argo_workflow_on_sqlflow.md]: SQL syntax error is the parsing phase, not workflow running phase

File: doc/design/argo_workflow_on_sqlflow.md [hunk 5]
```diff
@@ -186,3 +186,27 @@ where
...
+- The other is the workflow step runtime stage:
+  - Some workflow step execute a standard SQL, the GO database driver execute the SQL and return error if the execution failed.
+  - Some workflow step execute a extended SQL, SQLFlow translate the extended SQL into a submitter program and execute it
+    - Some submitter program would be executed as a sub-process, SQLFlow can retrieve the error logs from stderr.
+    - Some submitter program would run on a cluster system e.g. Yarn, SQLFlow should retrieve the error logs from the Yarn task.
+
+To create a good user experience, we should pipe theses error message to the GUI system like Jupyter Notebook.
+
+### How to pipe the error message
```
Comment (typhoonzero) [inline doc/design/argo_workflow_on_sqlflow.md]: How to Pipe the Error Message

File: doc/design/argo_workflow_on_sqlflow.md [hunk 6]
```diff
@@ -186,3 +186,27 @@ where
...
+
+``` text
+SQLFLow magic command(Jupyter Notebook) <----->  SQLFlow gRPC server  <---->  Workflow Step(Kubernetes Pod)
```
Comment (typhoonzero) [inline doc/design/argo_workflow_on_sqlflow.md]: Is SQLFlow server responsible for getting logs from logview?

File: doc/design/argo_workflow_on_sqlflow.md (hunk 2 above)
Comment (Yancey0623) [reply doc/design/argo_workflow_on_sqlflow.md]: I'm not sure, the training logs may be too long, which may make a bad user experience.

File: doc/design/argo_workflow_on_sqlflow.md (hunk 6 above)
Comment (Yancey0623) [reply doc/design/argo_workflow_on_sqlflow.md]: I think it's responsible for the workflow step, and work step should return the Longview URL to Notebook in the error message.

File: doc/design/argo_workflow_on_sqlflow.md (hunk 1 above)
Comment (typhoonzero) [reply doc/design/argo_workflow_on_sqlflow.md:190]: Seems still no endding "```" for above code snnipet

**Review by typhoonzero — DISMISSED** [review]

**Review by typhoonzero — APPROVED** [review]

**Pull request merged**